### PR TITLE
feat: add move errors to walrus-sui, remove clever errors

### PR DESCRIPTION
**Pull request by karlwuest**
This adds the error types but doesn't use them yet for the move calls.

### CODE REVIEW

File: crates/walrus-sui/src/types/move_errors.rs
Comment (mlegner) [inline crates/walrus-sui/src/types/move_errors.rs:1]: Question: Is there any way we can ensure this is in sync with the Move errors?

File: contracts/errors.out
Comment (mlegner) [inline contracts/errors.out:1]: Question: How is this generated and what is it used for?

File: contracts/modules.out
Comment (mlegner) [inline contracts/modules.out:1]: Question: How is this generated and what is it used for?

File: contracts/errors.out
Comment (karlwuest) [reply contracts/errors.out:1]: With `cat` and `grep` and a little bit of `vim` 😅 
Didn't mean to commit this (I thought `*.out` was in .gitignore). I will remove this (and modules.out).

File: crates/walrus-sui/src/types/move_errors.rs
Comment (karlwuest) [reply crates/walrus-sui/src/types/move_errors.rs:1]: I don't know of a good way to do this automatically. I added comments in the move files to keep the rust error codes up to date, but I agree that an automatic way would be best. I'll add an issue for this.

**Pull request merged**